frontend/passkeys.js

```diff
@@ -0,0 +1,31 @@
+document.addEventListener('DOMContentLoaded', function() {
+    'use strict';
+
+    const registerPasskeyButton = document.querySelector('#register-passkey');
+
+
+    async function registerPasskey(csrfToken) {
+        const response = await fetch('/passkey/registration-options') 
+        const publicKeyCredentialCreationOptionsJSON = await response.json() 
+        const credentialCreationOptions = PublicKeyCredential.parseCreationOptionsFromJSON(publicKeyCredentialCreationOptionsJSON); 
+        const publicKeyCredential = await navigator.credentials.create({publicKey: credentialCreationOptions}); 
+        const registrationResponseJSON = publicKeyCredential.toJSON(); 
+        await fetch('/passkey/register', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'text/plain',
+                'Csrf-Token': csrfToken
+            },
+            body: JSON.stringify(registrationResponseJSON)
+        });
+    }
+
+    registerPasskeyButton?.addEventListener('click', function(e) {
+            e.preventDefault();
+            const csrfToken = this.getAttribute('csrf-token');
+            registerPasskey(csrfToken).catch(function(err) {
+                console.error(err);
```

Comment:
Fine for now, but it'd be good to add some more robust error handling to help improve the user experience. In particular, our API endpoints return a nice JSON objec tthat contains a friendly user message, with the goal being that this is something we can display.

Again though, I think this is something to add later 👍 
today we're just putting these pages in to get an end-to-end test of the process so this looks ok to me.